### PR TITLE
Extract FileUtils updates by @ericvergnaud

### PR DESCRIPTION
**Pull request by parrt**
Extract FileUtils updates by @ericvergnaud in https://github.com/antlr/antlr4/pull/4027 then he can rebase after we merge this into dev.  All tests pass locally (didn't check python2 actually but python3 works).

Signed-off-by: Terence Parr <parrt@antlr.org>

### CODE REVIEW

**Comment by parrt** [comment]
@ericvergnaud ready for your review once tests pass. Then we can merge before ts.

File: runtime-testsuite/test/org/antlr/v4/test/runtime/FileUtils.java [hunk 1]
```diff
@@ -55,7 +61,7 @@ public static void mkdir(String dir) {
 	}
 
 	public static void deleteDirectory(File f) throws IOException {
-		if (f.isDirectory()) {
+		if (f.isDirectory() && !isLink(f.toPath())) {
```
Comment (KvanTTT) [inline runtime-testsuite/test/org/antlr/v4/test/runtime/FileUtils.java:64]: Could you clarify why link checking during deleting is important? What does it affect on?

Comment (ericvergnaud) [reply runtime-testsuite/test/org/antlr/v4/test/runtime/FileUtils.java:64]: During tests setup, the Antlr package is webpacked.
For each test, that package is installed as a dependency of the project in node_modules, using npm link.
Without this change, the antlr package gets deleted as part of destroying the test folder, which makes all subsequent tests fail.

**Comment by parrt** [comment]
During tests setup, the Antlr package is webpacked.
For each test, that package is installed as a dependency of the project in node_modules, using npm link.
Without this change, the antlr package gets deleted as part of destroying the test folder, which makes all subsequent tests fail.

**Pull request merged**